commit 083bcf9247bd8fa826927970f091325139d915a1 (refs/changes/50/150/1)
Author: Marsell Kukuljevic <marsell@joyent.com>
Date:   2016-07-27T15:58:16+10:00 (3 years, 2 months ago)
    
    PAPI-131 - deprecate the 'default' attribute, and remove any index
    on it. 'default' is still supported, but is now optional; it will be
    removed entirely in the future.

diff --git a/docs/index.md b/docs/index.md
index b863591..446572a 100644
--- a/docs/index.md
+++ b/docs/index.md
@@ -39,7 +39,6 @@ example of a package:
         uuid: "7fc87f43-2def-4e6f-9f8c-980b0385b36e",
         active: true,
         cpu_cap: 25,
-        default: false,
         group: "Standard",
         description: "Micro 0.25 GB RAM 0.125 CPUs 16 GB Disk",
         max_lwps: 4000,
@@ -64,7 +63,7 @@ example of a package:
 | [common_name](#package-common_name)                 |           |        |           | string  | Name displayed in the Portal.                                                                              |
 | cpu_burst_ratio                                     |           |        |           | float   | Typically computed value. See below for more.                                                              |
 | [cpu_cap](#package-cpu_cap)                         | sometimes |        | true      | integer | Cap on how much CPU a machine can use. 100 = one core, 350 = 3.5 cores, etc.                               |
-| [default](#package-default)                         | true      |        |           | boolean | Whether this is the default package of this name through the SDC 6.5 API                                   |
+| [default](#package-default)                         |           |        |           | boolean | **DEPRECATED** Whether this is the default package of this name through the SDC 6.5 API                    |
 | [description](#package-description)                 |           |        |           | string  | Human description of this package.                                                                         |
 | [fss](#package-fss)                                 |           |        |           | integer | CPU shares for a VM. This operates relative to other machines on a CN. (also known as cpu_shares)          |
 | [group](#package-group)                             |           |        |           | string  | Group of associated packages. E.g. High CPU, High Memory, High Storage, High IO or the customer's name.    |
@@ -122,11 +121,9 @@ cpu_cap is required by default, but can be made optional by setting
 
 ## Package: default
 
-Used for (old) packages requiring SDC6.5 compatibility. Several packages can
-have the same name, so the one which has default true is what is exposed through
-the SDC6.5. API.
+**DEPRECATED**
 
-    "default": true
+Was used for (old) packages requiring SDC6.5 compatibility.
 
 
 ## Package: description
diff --git a/etc/config.test.json b/etc/config.test.json
index 4bebdbe..707ebc7 100644
--- a/etc/config.test.json
+++ b/etc/config.test.json
@@ -9,7 +9,7 @@
             "minTimeout": 1000
         },
         "noCache": true,
-        "version": 5
+        "version": 6
     },
     "ufds_admin_uuid": "930896af-bf8c-48d4-885c-6573a94b1853",
     "first_boot": true,
@@ -69,8 +69,7 @@
         },
         "default": {
             "type": "boolean",
-            "index": true,
-            "required": true
+            "index": false
         },
         "group": {
             "type": "string",
diff --git a/sapi_manifests/papi/template b/sapi_manifests/papi/template
index db3d459..772dd0c 100644
--- a/sapi_manifests/papi/template
+++ b/sapi_manifests/papi/template
@@ -71,8 +71,7 @@
         },
         "default": {
             "type": "boolean",
-            "index": true,
-            "required": true
+            "index": false
         },
         "group": {
             "type": "string",
diff --git a/test/api.test.js b/test/api.test.js
index 7aa66ca..2796d08 100644
--- a/test/api.test.js
+++ b/test/api.test.js
@@ -303,7 +303,6 @@ test('POST /packages (missing required fields)', function (t) {
         var expectedErrs = [
             { field: 'active', code: 'Missing', message: 'is missing' },
             { field: 'cpu_cap', code: 'Missing', message: 'is missing' },
-            { field: 'default', code: 'Missing', message: 'is missing' },
             { field: 'max_lwps', code: 'Missing', message: 'is missing' },
             { field: 'max_physical_memory',
               code: 'Missing',
